hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/local/LocalRegionCache.java

```diff
@@ -195,18 +193,28 @@ protected Object createMessage(final Object key, final Object value, final Objec
     protected MessageListener<Object> createMessageListener() {
         return new MessageListener<Object>() {
             public void onMessage(final Message<Object> message) {
+                if (message.getPublishingMember().localMember()) {
+                    return;
+                }
                 final Invalidation invalidation = (Invalidation) message.getMessageObject();
-                if (versionComparator != null) {
-                    final Expirable value = cache.get(invalidation.getKey());
-                    if (value != null) {
-                        Object currentVersion = value.getVersion();
-                        Object newVersion = invalidation.getVersion();
-                        if (versionComparator.compare(newVersion, currentVersion) > 0) {
-                            cache.remove(invalidation.getKey(), value);
+                Object key = invalidation.getKey();
+                if (key != null) {
```

Comment:
So if the _key_ is `null` you clear the whole cache, where previously if the _comparator_ was `null` it would just remove the key.

`null`-checking the key is probably a good thing, but can you explain why you clear the entire cache on a `null` key?